packages/fxa-auth-server/test/remote/totp_tests.js

```diff
@@ -20,365 +20,332 @@ const {
-    return Client.login(config.publicUrl, email, password, {
-      ...testOptions,
-      keys: true,
-    }).then((response) => {
-      client = response;
-      assert.equal(
-        response.verificationMethod,
-        'totp-2fa',
-        'verification method set'
-      );
-      assert.equal(
-        response.verificationReason,
-        'login',
-        'verification reason set'
-      );
-
-      return client.requestVerifyEmail().then((res) => {
-        assert.deepEqual(res, {}, 'returns empty response');
-      });
     });
-  });
 
-  it('should not bypass `totp-2fa` by signing a cert with an unverified session', () => {
-    return Client.login(config.publicUrl, email, password, {
-      ...testOptions,
-      keys: false,
-    }).then((response) => {
-      client = response;
-      assert.equal(
-        response.verificationMethod,
-        'totp-2fa',
-        'verification method set'
-      );
-      assert.equal(
-        response.verificationReason,
-        'login',
-        'verification reason set'
-      );
+    it('should not have `totp-2fa` verification if user has unverified totp token', () => {
+      return client
+        .deleteTotpToken()
+        .then(() => client.createTotpToken())
+        .then(() =>
+          Client.login(config.publicUrl, email, password, {
+            ...testOptions,
+            keys: true,
+          })
+        )
+        .then((response) => {
+          assert.notEqual(
+            response.verificationMethod,
+            'totp-2fa',
+            'verification method not set to `totp-2fa`'
+          );
+          assert.equal(
+            response.verificationReason,
+            'login',
+            'verification reason set to `login`'
+          );
+        });
+    });
 
-      const publicKey = {
-        algorithm: 'RS',
-        n:
-          '4759385967235610503571494339196749614544606692567785790953934768202714280652973091341316862' +
-          '993582789079872007974809511698859885077002492642203267408776123',
-        e: '65537',
-      };
-      return client.sign(publicKey, 600).then(
-        () => {
-          assert.fail('should not have succeeded');
-        },
-        (err) => {
+    it('should not bypass `totp-2fa` by resending sign-in confirmation code', () => {
+      return Client.login(config.publicUrl, email, password, {
+        ...testOptions,
+        keys: true,
+      })
+        .then((response) => {
+          client = response;
           assert.equal(
-            err.errno,
-            138,
-            'should have failed due to unverified session'
+            response.verificationMethod,
+            'totp-2fa',
+            'verification method set'
           );
-        }
-      );
+          assert.equal(
+            response.verificationReason,
+            'login',
+            'verification reason set'
+          );
+
+          return client.requestVerifyEmail().then((res) => {
```

Comment:
This api now errors if a user attempts to send email confirmation code but has 2FA enabled. We previously return `{}`. I don't believe there are any consumers of this api besides us, so it is probably ok.